### PR TITLE
feat: user profile update

### PR DESCRIPTION
**Pull request by BiswaViraj**
### What change does this PR introduce?
- endpoint to get profile picture signed url
- e2e tests for signed url
- endpoint to update name and profile picture
- e2e tests for profile update

### CODE REVIEW

File: apps/api/src/app/storage/usecases/get-user-profile-picture-signed-url/get-user-profile-picture-signed-url.usecase.ts [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+  GetUserProfilePictureSignedUrlCommand,
+  MIME_TYPES_LOOKUP,
+} from './get-user-profile-picture-signed-url.command';
+
+@Injectable()
+export class GetUserProfilePictureSignedUrl {
+  constructor(private storageService: StorageService) {}
+
+  async execute(command: GetUserProfilePictureSignedUrlCommand): Promise<UploadUrlResponse> {
+    const path = `users/profile-pictures/${command.userId}/${hat()}.${command.extension}`;
```
Comment (BiswaViraj) [inline apps/api/src/app/storage/usecases/get-user-profile-picture-signed-url/get-user-profile-picture-signed-url.usecase.ts:16]: saves to folder path
`users -> profile-pictures -> userId -> image`

File: apps/api/src/app/user/usecases/update-name-and-profile-picture/update-name-and-profile-picture.command.ts [hunk 2]
```diff
@@ -0,0 +1,16 @@
...
+  @IsOptional()
+  imageUrl: string;
+
+  @IsDefined()
+  @IsString()
+  firstName: string;
+
+  @IsString()
+  @IsDefined()
+  lastName: string;
```
Comment (BiswaViraj) [inline apps/api/src/app/user/usecases/update-name-and-profile-picture/update-name-and-profile-picture.command.ts:15]: used firstName and lastName as we use similar properties for signup and also in user entity

File: apps/api/src/app/user/usecases/update-name-and-profile-picture/update-name-and-profile-picture.usecase.ts [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+      lastName: command.lastName,
+    };
+
+    const unsetPayload: Partial<Record<keyof UserEntity, string>> = {};
+
+    // if imageUrl exists, update the profile picture to the imageUrl, otherwise, unset/remove it
+    if (command.imageUrl) {
+      updatePayload.profilePicture = command.imageUrl;
+    } else {
+      unsetPayload.profilePicture = '';
```
Comment (BiswaViraj) [inline apps/api/src/app/user/usecases/update-name-and-profile-picture/update-name-and-profile-picture.usecase.ts:27]: removes old profilePicture if empty `imageUrl` passed

File: apps/api/src/app/storage/usecases/get-user-profile-picture-signed-url/get-user-profile-picture-signed-url.command.ts [hunk 4]
```diff
@@ -0,0 +1,19 @@
...
+  PNG = 'png',
+  JPG = 'jpg',
+}
+export class GetUserProfilePictureSignedUrlCommand extends EnvironmentWithUserCommand {
+  @IsString()
+  @IsEnum(MimeTypesEnum)
+  extension: string;
+}
+
+export const MIME_TYPES_LOOKUP = {
```
Comment (antonjoel82) [inline apps/api/src/app/storage/usecases/get-user-profile-picture-signed-url/get-user-profile-picture-signed-url.command.ts:15]:
```suggestion
export const MIME_TYPES_LOOKUP: Record<MimeTypesEnum, string> = {
```

**Review by antonjoel82 — APPROVED** [review]
I admittedly haven't done similar back-end work yet, but this looks nicely done and clean to me!

**Pull request merged**